ca/ca_test.go

```diff
@@ -0,0 +1,187 @@
+package ca
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/bwesterb/mtc"
+)
+
+// testCA is a Merkle Tree CA that runs in memory.
+type testCA struct {
+	*Handle
+	path string
+}
+
+// newTestCA creates a CA in a temporary directory. The caller should call
+// `Close()` in order to delete the directory once the test is finished.
+func newTestCA(t *testing.T) *testCA {
+	path, err := os.MkdirTemp("", "test-mtca-*")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	issuer := mtc.RelativeOID{}
+	err = issuer.UnmarshalText([]byte("1.2.3.4"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	handle, err := New(path, NewOpts{
+		Issuer:        issuer,
+		ServerPrefix:  "ca.example.com",
+		BatchDuration: time.Duration(0),
```

Comment:
I see division-by-zero errors in the future 😆. I was thinking of replacing all calls to real time, by a call to a virtual time function that can be passed when opening or creating the CA (just like the time function in crypto/tls), so that we can skip time in tests.

Reply:
Whoops, I was playing with this and meant to remove this line. Removed.

I had the same idea  but decided to do it in a separate PR. See the TODOs at the end of the file for some test cases for which this would be useful.